src/meltano/cli/state.py

```diff
@@ -310,12 +310,26 @@ def get_state(ctx: click.Context, project: Project, state_id: str) -> None:
 
 @meltano_state.command(cls=InstrumentedCmd, name="clear")
 @prompt_for_confirmation(prompt="This will clear state for the job. Continue?")
-@click.argument("state-id")
+@click.argument("state-id", required=False)
+@click.option(
+    "--all",
+    is_flag=True,
+    required=False,
+    help="Helps clear all states. Use with --force flag to skip confirmation.",
+)
 @pass_project(migrate=True)
 @click.pass_context
-def clear_state(ctx: click.Context, project: Project, state_id: str) -> None:
+def clear_state(ctx: click.Context, project: Project, state_id: str, all: bool) -> None:
     """Clear state."""
-    state_service: StateService = (
-        state_service_from_state_id(project, state_id) or ctx.obj[STATE_SERVICE_KEY]
-    )
-    state_service.clear_state(state_id)
+    if (not state_id and not all) or (
+        state_id and all
+    ):  # Case where neither or both have been provided
+        raise click.UsageError("Either a state ID or --all flag must be provided.")
```

Comment:
suggestion: either add test coverage for this. Or, since it's a fairly straightforward error state, don't count it

```suggestion
        raise click.UsageError("Either a state ID or --all flag must be provided.")  # pragma: no cover
```

Reply:
Done.